src/vs/platform/history/electron-main/historyMainService.ts

```diff
@@ -328,27 +346,38 @@ export class HistoryMainService implements IHistoryMainService {
 			// so we need to update our list of recent paths with the choice of the user to not add them again
 			// Also: Windows will not show our custom category at all if there is any entry which was removed
 			// by the user! See https://github.com/Microsoft/vscode/issues/15052
-			this.removeFromRecentlyOpened(app.getJumpListSettings().removedItems.filter(r => !!r.args).map(r => trim(r.args, '"')));
+			let toRemove: (ISingleFolderWorkspaceIdentifier | IWorkspaceIdentifier)[] = [];
+			for (let item of app.getJumpListSettings().removedItems) {
+				const args = item.args;
+				if (args) {
+					const match = /^--folderUri\s+"([^"]+)"$/.exec(args);
```

Comment:
@aeschli shouldn't it be `--folder-uri` to check for?!

Reply:
yes, good catch